lib/tasks/grade.rake

```diff
@@ -1,4 +1,223 @@
+# POTENTIAL IMPROVEMENTS
+# 1. Store personal token in ~/firstdraft.yml or Windows/Nitrous equivalent
+# 2. Task to update tests (perhaps automatically if checksum doesn't match)
+# 3. Associate pull request / commit to each submission
+# 4. Make 'descriptive' option more efficient
+
+# TESTING NEEDED
+# A. SSL & 'open' & 'rspec' commands on Windows, Nitrous
+
 desc "Grade project"
-task grade: :environment do
-  puts `bundle exec rspec --order default --format documentation --format j --out test_output.json --format html --out test_output.html`
+task :grade do # if needed in the future, add => :environment
+
+  options = {}
+  OptionParser.new do |opts|
+    opts.on("-v", "--verbose", "Show detailed test results") do |v|
+      options[:verbose] = v
+    end
+  end.parse!
+
+  begin
+    config_file_name_base = ".firstdraft_project.yml"
+    config_file_name = Rails.root.join(config_file_name_base)
+    config = YAML.load_file(config_file_name)
+    project_token = config["project_token"]
+    submission_url = config["submission_url"]
+  rescue
+    abort("ERROR: Does the file .firstdraft.yml exist?")
+  end
+  if !project_token
+    abort("ERROR: Is project_token set in .firstdraft.yml?")
+  end
+  if !submission_url
+    abort("ERROR: Is submission_url set in .firstdraft.yml?")
+  end
+
+  student_token_filename_base = ".firstdraft_student.yml"
+  gitignore_filename = Rails.root.join(".gitignore")
+  if File.readlines(gitignore_filename).grep(/^\/.firstdraft_student.yml$/).size == 0
```

Comment:
Eventually some of this more complex logic should probably be moved to helper methods.

There are probably some existing methods somewhere, possibly in [Thor](https://github.com/erikhuda/thor/wiki) (which is what Rails uses for generating files, etc).